### PR TITLE
fix(api): Add human friendly error messages for control and payload issues.

### PR DESCRIPTION
**Pull request by rifont**
### What changed? Why was the change needed?
* Add human friendly error messages for control and payload issues.

### Screenshots
_Editor canvas - better error message_
image

_Editor side panel - better error message_
image

<details>
<summary><strong>Expand for optional sections</strong></summary>

### Special notes for your reviewer

</details>

### CODE REVIEW

File: apps/api/src/app/workflows-v2/usecases/validate-content/prepare-and-validate-content/prepare-and-validate-content.usecase.ts [hunk 1]
```diff
@@ -198,7 +199,7 @@ export class PrepareAndValidateContentUsecase {
       result[item] = [
         {
           issueType: StepContentIssueEnum.MISSING_VALUE,
-          message: `[${item}] No Value was submitted to a required control.`,
+          message: `${capitalize(item)} is missing.`,
```
Comment (rifont) [inline apps/api/src/app/workflows-v2/usecases/validate-content/prepare-and-validate-content/prepare-and-validate-content.usecase.ts]: Yay or nay for the full stop? I couldn't decide which is cleaner. Deferring to a full stop for now. 🤔

File: apps/api/src/app/workflows-v2/usecases/validate-content/prepare-and-validate-content/prepare-and-validate-content.usecase.ts [hunk 2]
```diff
@@ -178,7 +179,7 @@ export class PrepareAndValidateContentUsecase {
           result[controlValueKey].push({
             issueType: StepContentIssueEnum.MISSING_VARIABLE_IN_PAYLOAD,
             variableName: item,
-            message: `[${item}] Missing Reference in payload`,
+            message: `${capitalize(item)} is missing in payload.`,
```
Comment (rifont) [inline apps/api/src/app/workflows-v2/usecases/validate-content/prepare-and-validate-content/prepare-and-validate-content.usecase.ts]:
```suggestion
            message: `${capitalize(item)} is missing in payload`,
```

File: apps/api/src/app/workflows-v2/usecases/validate-content/prepare-and-validate-content/prepare-and-validate-content.usecase.ts (hunk 1 above)
Comment (rifont) [reply apps/api/src/app/workflows-v2/usecases/validate-content/prepare-and-validate-content/prepare-and-validate-content.usecase.ts]: Designs show without the full stop. I've removed.

Comment (rifont) [inline apps/api/src/app/workflows-v2/usecases/validate-content/prepare-and-validate-content/prepare-and-validate-content.usecase.ts]:
```suggestion
          message: `${capitalize(item)} is missing`,
```

**Review by desiprisg — APPROVED** [review]

**Pull request merged**